contrib/scrooge/tests/python/pants_test/contrib/scrooge/tasks/test_scrooge_gen.py

```diff
@@ -80,6 +80,46 @@ def test_validate_compiler_configs(self):
         task._validate_compiler_configs(self.target("test_validate:one"))
         task._validate_compiler_configs(self.target("test_validate:two"))
 
+    def test_validate_unchecked_compiler_args(self):
+        # Set synthetic defaults for the global scope.
+        self.set_options_for_scope(
+            "thrift-defaults",
+            compiler="unchecked",
+            language="uniform",
+            service_deps="service_deps",
+            structs_deps="structs_deps",
+        )
+
+        self.add_to_build_file(
+            "test_validate",
+            dedent(
+                """
+                java_thrift_library(name='one',
+                  sources=[],
+                  dependencies=[],
+                )
+                """
+            ),
+        )
+
+        self.add_to_build_file(
+            "test_validate",
+            dedent(
+                """
+                java_thrift_library(name='two',
+                  sources=[],
+                  compiler_args=['--java-passthrough'],
+                  dependencies=[':one'],
+                )
+                """
+            ),
+        )
+
+        target = self.target("test_validate:two")
+        context = self.context(target_roots=[target])
+        task = self.prepare_execute(context)
+        task._validate_compiler_configs(self.target("test_validate:two"))
```

Comment:
Could we consider adding a failing test as well as a passing test? Or is the normal behavior (erroring out on different args) already tested elsewhere?

Reply:
Good suggestion. Thanks!